data-syncs/supported-data-sync-sources/rest-api.md

```diff
@@ -81,22 +90,121 @@ You can learn more about these sections in [Appendix A - Other Sections.](rest-a
 
 <figure><img src="../../.gitbook/assets/image (298).png" alt=""><figcaption><p>Image 2: The Source Tab</p></figcaption></figure>
 
+## Best practices
+
+### Retrieve nested fields
+
+To get fields in a nested array, you can either set the nested array as the root, or you can use Path to Iterate to expand the array. 
+
+#### Examples
+
+Here is a sample JSON response:
+
+```json
+[
+  {
+  "name": "John Doe",
+  "age": 30,
+  "hobbies": ["reading", "swimming", "coding"],
+  "contact": [{
+      "home": 555-1000,
+      "mobile": 555-2000
+  }
+  ]
+}
+]
+```
+
+
+Here
+### JSON array handling
+
+Use `$.data` in **Records Root JSONPath** if the API returns a top-level JSON array.
+
+#### Examples
+
+Here is a sample JSON response:
+
+```json
+[
+  {
+  "name": "John Doe",
+  "age": 30,
+  "hobbies": ["reading", "swimming", "coding"],
+  "contact": [{
+      "home": 555-1000,
+      "mobile": 555-2000
+  }
+  ]
+}
+]
+```
+
+If you use `$` as the **Records Root JSONPath** and the top level is an array, then you must append `$.data` to each schema. For example, for the JSON array above, you can do either of the following:
+
+### Using "Data" as root
+
+This example uses the JSON response above, and wants to map two columns in a Cinchy Table, **Name** and **Age**, to the keys `"Name"` and `"Age"` in the JSON response.
+
+**Records Root JSONPath**: `$.data`
+
+**Schema**:
+
+- `$.name` for **Name**
+- `$.age` for **Age**
+
+#### Using $ as root
+
+**Records Root JSONPath**: `$`
+
+**Schema**:
+
+- `$.data.name` for **Name**
+- `$.data.age` for **Age**
+
+#### Example
+### Path to Iterate
+
+Use Path to Iterate to expand records within an array. It allows you to target nested keys within the array. This only applies if the records within an array are objects.
+
+If the record within the path to iterate is an array, it will be wrapped in an `"item"` object.
```

Comment:
each item within the  array gets placed under an "item" key in a new JSON object.